app.rb

```diff
@@ -0,0 +1,116 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book_manager'
+require_relative 'person_manager'
+require_relative 'rental_manager'
+class App
+  def initialize
+    @book_manager = BookManager.new
+    @person_manager = PersonManager.new
+    @rental_manager = RentalManager.new
+  end
+
+  def list_books
+    puts 'List of Books:'
+    @books.each_with_index { |book, index| puts "#{index}) #{book.title} by #{book.author}" }
+  end
+
+  def list_people
+    puts 'List of People:'
+    @people.each do |person|
+      if person.is_a?(Teacher)
+        puts "#{person.name} (Teacher, Age: #{person.age}, Specialization: #{person.specialization})"
+      else
+        puts "#{person.name} (Student)"
+      end
+    end
+  end
```

Comment:
- [ ] Hello! 👋 While it's great that you've implemented methods for listing books and people in your app.rb file, I've noticed that these same methods are also redundantly defined in book_manager.rb and person_manager.rb. 📚👥 This redundancy can lead to confusion and can potentially cause issues within the app. 😕 It's important to keep two key principles in mind: KISS (Keep It Simple, Stupid) and YAGNI (You Ain't Gonna Need It).

![Screenshot from 2023-10-26 13-15-48](https://github.com/faisalakb/Ruby_OOP_school_library/assets/112261237/8fe85a36-df0e-4301-bb8b-c784180cebd9)